flink-table/flink-table-common/src/main/java/org/apache/flink/table/catalog/AbstractCatalog.java

```diff
@@ -47,4 +51,31 @@ public String getName() {
     public String getDefaultDatabase() {
         return defaultDatabase;
     }
+
+    @Override
+    public String explainCatalog() {
+        StringBuilder sb = new StringBuilder();
+        Map<String, String> extraExplainInfo = extraExplainInfo();
+        sb.append("default database: ").append(defaultDatabase);
+        if (!extraExplainInfo.isEmpty()) {
+            sb.append("\n");
+            sb.append(
+                    extraExplainInfo.entrySet().stream()
+                            .map(entry -> entry.getKey() + ": " + entry.getValue())
+                            .collect(Collectors.joining("\n")));
+        }
+
+        // put the class name at the end
+        sb.append("\n").append(Catalog.super.explainCatalog());
+        return sb.toString();
+    }
+
+    /**
+     * Extra explain information used to print by DESCRIBE CATALOG catalogName statement.
+     *
+     * <p>Note: The class name and default database name of this catalog are no need to be added.
+     */
+    protected Map<String, String> extraExplainInfo() {
```

Comment:
Can we directly retrieve parameter information from the catalog store instead of introducing a new interface?